indico/core/signals/event/registration.py

```diff
@@ -18,7 +18,8 @@
 registration_state_updated = _signals.signal('registration-state-updated', '''
 Called when the state of a registration changes. The `sender` is the
 `Registration` object; the previous state is passed in the `previous_state`
-kwarg.
+kwarg; whether the transition should be logged is passed in the `log` kwarg,
+which defaults to `True`.
```

Comment:
There is no such thing as a default value for a signal kwarg - if an arg is not specified, it's not there. Technically your implementation works, because in the place where it matters your signal handler sets a default value. But for anything else handling that signal, the kwarg won't be there unless the code triggering the signal sends it (which is usually not the case).

This is why I suggested something like `no_log`: When it's not there, it's the normal case, and in the special case (kwarg present and set to True) there's special logic (no logging).

Reply:
How about `silent=True` if you don't like `no_log`?